packages/jobs/src/job-definitions/batchEvaluations/runDocumentJob.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  Commit,
+  DocumentVersion,
+  LogSources,
+} from '@latitude-data/core/browser'
+import { jobs } from '@latitude-data/core/jobs'
+import { connection } from '@latitude-data/core/redis'
+import { runDocumentAtCommit } from '@latitude-data/core/services/commits/runDocumentAtCommit'
+import { env } from '@latitude-data/env'
+import { Job } from 'bullmq'
+
+import { ProgressTracker } from '../../utils/progressTracker'
+
+type RunDocumentJobData = {
+  workspaceId: number
+  document: DocumentVersion
+  commit: Commit
+  parameters: Record<string, unknown>
+  evaluationId: number
+  batchId: string
+}
+
+export const runDocumentJob = async (job: Job<RunDocumentJobData>) => {
+  const { workspaceId, document, commit, parameters, evaluationId, batchId } =
+    job.data
+
+  const progressTracker = new ProgressTracker(connection, batchId)
+
+  try {
```

Comment:
Why try/catch if the service is using a result?

Reply:
there's 3 awaits that could throw in here, i wanna make sure we handle those